addons/mail/static/src/messaging/entity/follower/follower.js

```diff
@@ -0,0 +1,207 @@
+odoo.define('mail.messaging.entity.Follower', function (require) {
+'use strict';
+
+const {
+    fields: {
+        many2many,
+        many2one,
+    },
+    registerNewEntity,
+} = require('mail.messaging.entity.core');
+
+function FollowerFactory({ Entity }) {
+
+    class Follower extends Entity {
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @param {mail.messaging.entity.FollowerSubtype} subtype
+         */
+        checkSubtype(subtype) {
+            if (!this.checkedSubtypes.includes(subtype)) {
+                this.link({ checkedSubtypes: subtype });
+            }
+        }
+
+        closeSubtypes() {
+            this.env.messaging.dialogManager.close(this._subtypesListDialog);
+            this._subtypesListDialog = undefined;
+        }
+
+        /**
+         * @return {string}
+         */
+        get name() {
+            if (this.channel) {
+                return this.channel.name;
+            }
+            return this.partner.name;
+        }
+
+        /**
+         * Remove this follower from its related thread.
+         */
+        async remove() {
+            await this.env.rpc({
+                model: this.followedThread.model,
+                method: 'message_unsubscribe',
+                args: [
+                    [this.followedThread.id],
+                    this.partner ? [this.partner.id] : [],
+                    this.channel ? [this.channel.id] : [],
```

Comment:
I would change this similar to `message_subscribe`, using kwargs and only sending the relevant information.